DependencyInjection/Compiler/DoctrineDBALLoggerPass.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\DBAL\Logging\SQLLogger;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class DoctrineDBALLoggerPass implements CompilerPassInterface
+{
+    private const TAG_NAME = 'doctrine.dbal.logger';
+    private const BASE_CHAIN_NAME = 'doctrine.dbal.logger.chain';
+
+    public function process(ContainerBuilder $container)
+    {
+        $serviceList = $container->findTaggedServiceIds(self::TAG_NAME, true);
+        $serviceList = array_keys($serviceList);
+
+        if (empty($serviceList)) {
+            return;
+        }
+
+        foreach ($serviceList as $serviceId) {
+            $this->ensureIsValidServiceDefinition($container, $serviceId);
+        }
+
+        $connectionList = $container->getParameter('doctrine.connections');
+
+        foreach ($connectionList as $connectionName => $connectionId) {
+            $configurationDefinition = $container->getDefinition($connectionId . '.configuration');
+            $chainDefinition = $this->createChainDefinitionAttachedToConnection($container, $connectionName, $configurationDefinition);
+
+            foreach ($serviceList as $serviceId) {
+                $serviceReference = new Reference($serviceId);
+                $chainDefinition->addMethodCall('addLogger', [$serviceReference]);
+            }
+        }
+    }
+
+    private function ensureIsValidServiceDefinition(ContainerBuilder $container, string $serviceId): void
```

Comment:
Is this needed?

I have never noticed any other doing this much for detecting miss configurations of services and symfony it self is happy to fail when you try to retrieve the service from the container.

Reply:
Anyone have a comment on this?

We don't have this anywhere else neither have i seen others do it.

Reply:
Yep it's not needed